### PR TITLE
game board component

### PR DESCRIPTION
**Pull request by seshunathk**
Game Board UI Component. 

Screen Shot 2021-04-19 at 10 40 17 PM

### CODE REVIEW

File: client/src/pages/GameBoard.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import { makeStyles } from '@material-ui/core/styles';
+import { Grid, Paper } from '@material-ui/core';
+
+const useStyles = makeStyles({
+    root: {
+      flexGrow: 1,
+      marginTop: '20px',
+      display: 'grid',
+      gridTemplateColumns: 'auto auto auto auto auto'
+    },
```
Comment (dproc96) [inline client/src/pages/GameBoard.js]: It's not great practice to redefine the grid properties of Material UI but in this case I'm not sure there's a not-hacky way to get five columns. If Jess and your team are okay with it then I think this is fine

**Review by dproc96 — COMMENTED** [review]
Had one comment for you to look into, otherwise the UI looks good

File: client/src/pages/GameBoard.js (hunk 1 above)
Comment (jessdesr) [reply client/src/pages/GameBoard.js]: I think it would make sense to use Box components instead of combining both Flexbox and CSS Grid

Comment (seshunathk) [reply client/src/pages/GameBoard.js]: I have changed to the box component and implemented onClick functionality on each card then the color changes to grey.

Screen Shot 2021-04-21 at 11 48 51 AM

**Comment by CSimbulan** [comment]
Added styling based on what kind of card it is: 
 
Guesser cards: 
image

Spymaster view:
image

Revealed cards:
image

Onclick functionality requires backend integration, so that will be added in the future.
The code to randomize the board is just copied from the game engine for testing, so it will be removed during integration.

**Comment by CSimbulan** [comment]
Since the game layout was already merged to the dev branch I just added the board component onto it. 

image

**Comment by CSimbulan** [comment]
image

File: client/src/components/GameBoard.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    return (
+        <div className={classes.root}>
+            <div className={classes.container}>
+                {board.map((function (row) {
+                    return row.map(
+                        card => (
+                            // Spymaster view coded in as a prop for now, will change during integration.
+                            <Card key={card.word} word={card.word} color={card.color} spyMaster={false} revealed={!card.revealed} />
```
Comment (dproc96) [inline client/src/components/GameBoard.js]:
```suggestion
                            <Card key={card.word} word={card.word} color={card.color} spyMaster={false} revealed={card.revealed} />
```
Is this set this way for testing right now? You can probably change it before merging so it doesn't cause issues later

File: client/src/components/Card.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+    return (
+
+        <Box className={
+            (props.revealed ? classes[props.color + "Revealed"] + " " + classes["whiteTextRevealed"] : (
+                props.spyMaster ? classes[props.color + "Text"] : ""
+            ) + " " + classes["defaultHidden"]) + " " + classes["box"]}>
+            {props.word}
+        </Box>
+    );
```
Comment (dproc96) [inline client/src/components/Card.js]: Can you clean up the spacing here? Prettier should be able to do it easily

**Review by dproc96 — APPROVED** [review]
Two small change requests and can you address the warnings in the console, but once those are done this is good to merge

**Pull request merged**